tests/rc_string_ut/rc_string_ut.c

```diff
@@ -26,40 +42,432 @@ static void on_umock_c_error(UMOCK_C_ERROR_CODE error_code)
+}
+
+TEST_SUITE_CLEANUP(suite_cleanup)
+{
+    umock_c_deinit();
 
-        ASSERT_ARE_EQUAL(int, 0, umock_c_init(on_umock_c_error));
-        ASSERT_ARE_EQUAL(int, 0, umocktypes_charptr_register_types());
+    TEST_MUTEX_DESTROY(g_testByTest);
+}
+
+TEST_FUNCTION_INITIALIZE(function_init)
+{
+    if (TEST_MUTEX_ACQUIRE(g_testByTest))
+    {
+        ASSERT_FAIL("our mutex is ABANDONED. Failure in test framework");
     }
 
-    TEST_SUITE_CLEANUP(suite_cleanup)
+    ASSERT_ARE_EQUAL(int, 0, umock_c_negative_tests_init());
+
+    umock_c_reset_all_calls();
+}
+
+TEST_FUNCTION_CLEANUP(function_cleanup)
+{
+    umock_c_negative_tests_deinit();
+    TEST_MUTEX_RELEASE(g_testByTest);
+}
+
+/* rc_string_create */
+
+/*Tests_SRS_RC_STRING_01_001: [ If string is NULL, rc_string_create shall fail and return NULL. ]*/
+TEST_FUNCTION(rc_string_create_with_NULL_fails)
+{
+    // arrange
+
+    // act
+    THANDLE(RC_STRING) rc_string = rc_string_create(NULL);
+
+    // assert
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+    ASSERT_IS_NULL(rc_string);
+}
+
+/* Tests_SRS_RC_STRING_01_002: [ Otherwise, rc_string_create shall determine the length of string. ]*/
+/* Tests_SRS_RC_STRING_01_003: [ rc_string_create shall allocate memory for the THANDLE(RC_STRING), ensuring all the bytes in string can be copied (including the zero terminator). ]*/
+/* Tests_SRS_RC_STRING_01_004: [ rc_string_create shall copy the string memory (including the NULL terminator). ]*/
+/* Tests_SRS_RC_STRING_01_005: [ rc_string_create shall succeed and return a non-NULL handle. ]*/
+TEST_FUNCTION(rc_string_create_succeeds)
+{
+    // arrange
+    STRICT_EXPECTED_CALL(malloc(IGNORED_ARG));
+
+    // act
+    THANDLE(RC_STRING) rc_string = rc_string_create("gogu");
+
+    // assert
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+    ASSERT_ARE_EQUAL(char_ptr, "gogu", rc_string->string);
+    ASSERT_IS_NOT_NULL(rc_string);
+
+    // cleanup
+    THANDLE_DEC_REF(RC_STRING)(rc_string);
+}
+
+/* Tests_SRS_RC_STRING_01_002: [ Otherwise, rc_string_create shall determine the length of string. ]*/
+/* Tests_SRS_RC_STRING_01_003: [ rc_string_create shall allocate memory for the THANDLE(RC_STRING), ensuring all the bytes in string can be copied (including the zero terminator). ]*/
+/* Tests_SRS_RC_STRING_01_004: [ rc_string_create shall copy the string memory (including the NULL terminator). ]*/
+/* Tests_SRS_RC_STRING_01_005: [ rc_string_create shall succeed and return a non-NULL handle. ]*/
+TEST_FUNCTION(rc_string_create_with_empty_string_succeeds)
+{
+    // arrange
+    STRICT_EXPECTED_CALL(malloc(IGNORED_ARG));
+
+    // act
+    THANDLE(RC_STRING) rc_string = rc_string_create("");
+
+    // assert
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+    ASSERT_ARE_EQUAL(char_ptr, "", rc_string->string);
+    ASSERT_IS_NOT_NULL(rc_string);
```

Comment:
>ASSERT_IS_NOT_NULL(rc_string); [](start = 4, length = 30)

switch lines 147 and 148

Reply:
Switched

---
In reply to: [460260660](https://github.com/Azure/azure-c-util/pull/52#discussion_r460260660) [](ancestors = 460260660)